shell/ci/release/confluence-publish.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# This script publishes all markdown files with certain directives to confluence
+# using said directives.
+#
+# Directives:
+# <!-- Space: <confluence space key> --> (required)
+# <!-- Parent: <title of parent page> --> (not required)
+# <!-- Title: <title of page to publish to> --> (required)
```

Comment:
First take: Not a huge fan of having a custom DSL in bash here, but it's fine. We can always write go here, the e2e runner is an example of that.